scripts/data_generation.py

```
@@ -0,0 +1,344 @@
+                    response_format={"type": "json_object"}
+                )
+                
+                # Extract and parse the responses
+                content = response.choices[0].message.content
+                try:
+                    response_data = json.loads(content)
+                    if "responses" in response_data:
+                        return response_data["responses"]
+                    else:
+                        # Try to find an array in the response
+                        for key, value in response_data.items():
```

Ruff (0.14.1)

Loop control variable `key` not used within loop body

Rename unused `key` to `_key`

(B007)